File: services/python/starlette/src/routes/handlers/invalid_path.py
import structlog
from starlette.requests import Request
from starlette.responses import Response

from routes.errors import AppErrors, send_error


async def invalid_path_handler(request: Request, _exception: Exception) -> Response:
    structlog.get_logger().warn(
        "Invalid path", method=request.method, path=request.url.path
    )

    return send_error(AppErrors.InvalidPath)

File: services/go/gin/routes/handlers/invalid_path.go
package handlers

import (
	"gin/routes"
	"log/slog"

	"github.com/gin-gonic/gin"
)

func InvalidPathHandler(c *gin.Context) {
	slog.Warn("Invalid path", "method", c.Request.Method, "path", c.Request.URL.Path)

	routes.SendError(c, routes.AppErrors.InvalidPath)
}
